lib/yabeda/dsl/class_methods.rb

```diff
@@ -58,8 +58,12 @@ def histogram(*args, **kwargs, &block)
       #
       # @param name [Symbol] Name of default tag
       # @param value [String] Value of default tag
-      def default_tag(name, value)
-        ::Yabeda.default_tags[name] = value
+      def default_tag(name, value, group: nil)
```

Comment:
You can use this and it will work :smiley: 
```suggestion
      def default_tag(name, value, group: @group)
```

Reply:
I've pushed up a new commit (to maybe make reviewing easier). I'll squash prior to merge if it all looks good.

Some thoughts as I was working onit: 
1. I created a group object with a `name` of `nil` to account for the global group.
1. `with_tags` can nest `with_tags`
1. Named `group`'s continue to be a child of the global group. Current code doesn't support deeper nesting, and `Yabeda.group1.group2.some_gauge.set({}, 5)` feels verbose/complicated (and probably not needed).
1. I got into some weird states on testing when trying to remove methods so I tried to make `Yabeda.reset!` a bit more resilient and comprehensive (so a bad test doesn't break every subsequent test).
1. I got the feeling that a group instance could be the evaluation context for the DSL. A group having default_tags and having metrics already feels very similar to the `Yabeda` module. If this was done, then arbitrary nesting of groups becomes easier. 
1. The two register calls for groups and metrics was a bit confusing since they both call each other.